### ISSUE TITLE
3.0.7 interferes with (QNote++) plugins

### ISSUE THREAD

**Issue by DrBoym** [opto/nostalgy-xpi#128]
Hi there,

running TB 91.0.3, Nostalgy 3.0.7 and QNote++ 0.10.2 I end up with Nostalgy Hotkeys being active when editing a QNote++ entry for a mail or such. 
I expect that the same could happen with other plugins, not tested yet.
Workarounds: 
A) Restrict command to email tabs works, but only after setting the option and restarting TB.
B) Define hotkeys that are not used for typing text

Related to that is another issue: Hotkeys in Nostaly cannot be disabled anymore: Click on disable in Key menu does not have any effect what so ever.

**Comment by opto** [comment]
yes, A is the recommended solution for that, because at the moment, after a restartm TB does not recognise the mail window correctly.

**Comment by opto** [comment]
the second issue is a dublicate of #129

**Comment by DrBoym** [comment]
> yes, A is the recommended solution for that, because at the moment, after a restartm TB does not recognise the mail window correctly.

I am afraid in 3.0.8 but also 3.0.7 "restrict command to email tabs" does not solve this issue consistently for plugins like QNote. Thus this issue should probably not be closed as it requires a fix still.

**Comment by opto** [comment]
is the qnote edited in the main mail:3pane window? If yes, this will be diffucult to change, because normally, nobody edits in that window.

Can you reset the keys in options to something not used in the note?

**Comment by DrBoym** [comment]
> is the qnote edited in the main mail:3pane window? If yes, this will be diffucult to change, because normally, nobody edits in that window.
I am not 100% sure, but I would say no. It is an overlay that you edit in. Functionality is basically exactly the same as in [XNote++](https://github.com/xnotepp/xnote), which I know you know way better than me @opto ;-). 
I only  switched to QNote (really sorry for that!) a while back when TB 78 was released, as XNote didn't work for quiet a while and I had to get my old notes back in order to be able to work. QNote++ is capable of working on old notes taken with XNote or XNote++. 
Current XNote++  v3.2.1 does not show the bug described here. Typing text including a letter defined as a hotkey does not trigger Nostalgy, as it does in QNote++ currently. 

**Therefore, there is this workaround:**
I switched to use XNote++ again, but keep QNote++ running in parallel just to read my old QNotes with that. Of cause doing this is a bit annoying when you hit a mail with a new XNote++ it will show up twice, one note by XNote++ and the same one in a different GUI from QNote. 
Unfortunately XNote++ is not capable of reading/editing old QNote++ notes. Only the other way round...QNote is able to read XNote notes ^^
Maybe it is a nice feature idea for XNote++ to make it compatible with QNote notes? This way users could define the same storage path for notes in both plugins and work with that one, which is currently working best based on the TB Version used and user preference of cause. 
Wait... Yes, it would be nice, as others mentioned earlier ;-): https://github.com/xnotepp/xnote/issues/62

> Can you reset the keys in options to something not used in the note?
In theory yes. But as you require all characters and numbers as well as symbols to write a meaningful note it is quiet difficult to define a set of useful hotkeys for Nostalgy.

**Comment by ThomasHeinrichSchmidt** [comment]
Version 3.2.16 of Nostalgy++ released (on TB 91.6.0) but this bug is still not corrected. I'm still not able to type QNotes while Nostalgy++ is active.

**Comment by ThomasHeinrichSchmidt** [comment]
It seems that changes in keystroke detection (perhaps in N++ or in TB or both) also have an impact on several other issues, like #121 #122, #127, #130

**Comment by opto** [comment]
see first post:
A) Restrict command to email tabs works, but only after setting the option and restarting TB.

does that help? It should be the recommended way and is set as standard now, but old installations don't pick it up.

Klaus

**Comment by opto** [comment]
@[ThomasHeinrichSchmidt](https://github.com/ThomasHeinrichSchmidt)

> It seems that changes in keystroke detection (perhaps in N++ or in TB or both) also have an impact on several other issues, like #121 #122, #127, #130

#122, #127 is resolved, not working for you?

**Comment by ThomasHeinrichSchmidt** [comment]
I'm using "Restrict commands to email tab" all the time.

![Nostalgy-Preferences](https://user-images.githubusercontent.com/8068047/154037221-e51efd68-2493-496f-9a26-363391ac1cc6.jpg)

But as soon as I type a N++ command letter while editing a QNote (e.g. `p q r s`) N++ jumps in

![QNote-typing-PQRS](https://user-images.githubusercontent.com/8068047/154037455-e8e984bb-b3a1-427d-af1f-4c16d3474881.jpg)

**Comment by Olifair** [comment]
> @[ThomasHeinrichSchmidt](https://github.com/ThomasHeinrichSchmidt)
> 
> > It seems that changes in keystroke detection (perhaps in N++ or in TB or both) also have an impact on several other issues, like #121 #122, #127, #130
> 
> #122, #127 is resolved, not working for you?

and #130 was also solved (for me) since 3.1.3 :-)

**Comment by ThomasHeinrichSchmidt** [comment]
I saw references to these issues when I read about problems similar to mine. Good if most of those problems have been resolved. But still I cannot type into QNote as I could before TB 91 and Nostalgy++ 3.0.7.

**Comment by SergeLalonde** [comment]
I still have the same problem as @ThomasHeinrichSchmidt. The Restrict commands to email tab is on and makes no difference. I tried unchecking it, restarting, rechecking, restarting and there's no change.
Is there really no fix for this? I installed XNote++ and it works fine. Unfortunately, I can't switch to it since it doesn't see all of the QNotes that I've already created. Keeping both installed is clumsy.

**Comment by ThomasHeinrichSchmidt** [comment]
In the end I tried to disable most of the N++ keys leaving just Ctrl+S and Shift+Ctrl+S for "Save message" and "Save as suggested" respectively. But this does not work either because N++ does not save the "disable" option. When I come back to the "Nostalgy Preferences" dialog all the keys are enabled again.

**Comment by mlazdans** [comment]
Please add this on check to onNostalgyKeyPress(ev) function. I can not send merge request because this repo seems out to date compared with just released 3.3.19

```
...
function onNostalgyKeyPress(ev) {
  if(ev.target && ev.target.className == 'webextension-popup-browser'){
    return;
  }
....
```
